WcaOnRails/app/assets/javascripts/associated-events-picker.js

```diff
@@ -1,10 +1,24 @@
 $(function() {
   function checkboxesSetter(value) {
     return function() {
-      $(this).closest('.form-group').find('.event-checkbox input[type="checkbox"]').prop('checked', value);
+      var $eventsFormGroup = $(this).closest('.form-group');
+      $eventsFormGroup.find('.event-checkbox input[type="checkbox"]').prop('checked', value);
+      updateEventsSelectedCount($eventsFormGroup);
     };
   }
 
   $('.select-all-events').on('click', checkboxesSetter(true));
   $('.clear-all-events').on('click', checkboxesSetter(false));
+
+  function updateEventsSelectedCount($eventsFormGroup) {
+    var count = $eventsFormGroup.find('input[type="checkbox"]:checked').size();
+    $eventsFormGroup.find('.associated-events-label .events-selected-count').text(count);
+  }
+
+  $('.associated-events').on('change', function(e) {
+    updateEventsSelectedCount($(this).closest('.form-group'));
+  });
+  $('.associated-events').each(function() {
+    updateEventsSelectedCount($(this).closest('.form-group'));
+  });
```

Comment:
Neither of those fire on page load. I suspect that's because the ready/load event fires on the document body and only bubbles upwards, not downwards.

Here's a silly hack to achieve what you want. What do you think?

```
  $('.associated-events').on('change', function(e) {
    updateEventsSelectedCount($(this).closest('.form-group'));
  }).trigger('change');
```